### PR TITLE
Ubuntu 18.04 & Tokens

### PR DESCRIPTION
**Pull request by onvmstats**
Ubuntu 18.04 is now where CI lives, and Personal Access Tokens are in use to be in accordance with Github's deprecation notice of user/pass hooks.

## Summary:
Since moving the CI server to Ubuntu 18.04, there were a few changes necessary. We now use a specific CI user to run CI on the server instead of my account (which just makes more sense), so a new field `worker_user` was required for the scripted server to understand. For posterity reasons, the README was significantly updated so someone other than me will know how to move CI in the future to a new server if necessary. In some of the python scripts, we needed to run the paramiko logic in a separate function, which I called `paramiko_run` because there were issues with global variables being used. Finally, the only new script added was a python script using github3.py's token creation system. The configs have been updated to use the token, instead of just username and password. This was requested by Github as a deprecation notice. A final small change to `run_ci.sh` was adding a new install package required called `pycryptodome`. This was left out previously in the install guide, but is required to run correctly upon fresh install.

**Usage:**
CI will run just as usual, it is more secure now with the tokens, though.

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                |
| New functionality        | 👍 
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: None  

**TODO before merging :**
 - [ ] PR is ready for review

## Test Plan:
Run CI on a few different PRs to ensure that it is working correctly. @nks5295 and I will keep in touch regarding the tokens, to ensure the permissions are secure and cannot be taken advantage of.

## Review: 
@koolzz @dennisafa @nks5295

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

File: ci/post-msg.py [hunk 1]
```diff
@@ -36,17 +36,16 @@ def add_results_from_worker(file):
...
-        previous_results_from = results['results_from'] 
+        previous_results_from = results['results_from']
     comment_body += "\n" + results['summary']
 
 with open(sys.argv[1], "r") as credsfile:
     creds = [x.strip() for x in credsfile.readlines()]
 
-if len(creds) != 2:
+if len(creds) != 3:
     print("ERROR: Incorrect number of lines in credentials file!")
```
Comment (dennisafa) [inline ci/post-msg.py]: Should we sys_exit here as well? I notice that we do it in run_workload and prepare_worker.

File: ci/clone-and-checkout-pr.py [hunk 2]
```diff
@@ -15,15 +15,14 @@
 with open(sys.argv[1], "r") as credsfile:
     creds = [x.strip() for x in credsfile.readlines()]
 
-if len(creds) != 2:
+if len(creds) != 3:
     print("ERROR: Incorrect number of lines in credentials file!")
```
Comment (dennisafa) [inline ci/clone-and-checkout-pr.py]: Same here

**Comment by koolzz** [comment]
@kevindweb I'm a bit lost on the status of this one, lmk what the plan is

**Comment by kevindweb** [comment]
I already started moving CI over to the new repository. I think we just ignore this change. After CI is completely updated and on the other repo (and in a VM), I'll make a PR deleting CI from this repo. How does that sound? At that point I'll remove this PR so for now ignore this PR.

**Comment by onvmstats** [comment]
@onvm just run linter

**Comment by onvm** [comment]
> @onvm just run linter

## CI Message

Your results will arrive shortly

**Comment by onvm** [comment]
> @onvm just run linter

## CI Message

Error: ERROR: Failed to fetch and checkout pull request

**Comment by onvmstats** [comment]
@onvm testing

**Comment by onvm** [comment]
> @onvm testing

## CI Message

Your results will arrive shortly

**Comment by onvmstats** [comment]
@onvm testing with VM service

**Comment by onvmstats** [comment]
@onvm another test

**Comment by onvmstats** [comment]
@onvm test again

**Comment by onvmstats** [comment]
@onvm let's test linter

**Comment by onvm** [comment]
> @onvm let's test linter

## CI Message

Your results will arrive shortly

**Comment by kevindweb** [comment]
@onvm Run please

**Comment by onvm** [comment]
> @onvm Run please

## CI Message

Your results will arrive shortly

**Comment by kevindweb** [comment]
@onvm test with cleaned code

**Comment by onvm** [comment]
> @onvm test with cleaned code

## CI Message

Your results will arrive shortly

**Comment by kevindweb** [comment]
I will be making a new pull request for removing CI from onvm, goodbye old friend  🖖🏽, Vulcan salute to you!

**Pull request closed without merging**